### PR TITLE
PERF-3656 Modify ContentionTTLDeletions to simulate more contention

### PR DESCRIPTION
**Pull request by haleyConnelly**

### CODE REVIEW

**Comment by haleyConnelly** [comment]
We noticed that the existing test was not actually leading to consistent contention (write out tickets always greater than 0). This change introduces CRUD operations to a separate collection, along with heavy write workload to a TTL collection. 

It also serves as a measurement of performance for the Deprioritization project ([PM-2935](https://jira.mongodb.org/browse/PM-2935)) as follows:

- When the deprioritisation is enabled, CRUD operation throughput should be increased at the expense of a decreased amount of TTL document deletions

**Performance History Tracking**
- There is an expected decline in "performance" measurements with these modifications, a new baseline will be established for this test once the change is committed, as it significantly increases contention

[Evergreen](https://spruce.mongodb.com/version/63a245c5d1fe074ba163f0ee/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)

File: src/workloads/scale/ContentionTTLDeletions.yml [hunk 1]
```diff
@@ -51,7 +39,7 @@ Actors:
   Phases:
     OnlyActiveInPhases:
       Active: [1]
```
Comment (brad-devlugt) [inline src/workloads/scale/ContentionTTLDeletions.yml:41]: Do you want these admin commands to run before the data is loaded? Currently they are both set to run in Phase 1, so they will run in parallel.

File: src/workloads/scale/ContentionTTLDeletions.yml [hunk 2]
```diff
@@ -14,35 +14,23 @@ Keywords:
...
+  counter: 0
+  expireDate: {^Date: "1970-01-01"}
+
+LargeDoc: &LargeDoc
+  a: {^FastRandomString: {length: 67890}}
+  numField: {^RandomInt: {min: 0, max: 1000}}
+  counter: 0
+  expireDate: {^Date: "1970-01-01"}
 
+WMajorityJFalse: &WMajorityJFalse
```
Comment (brad-devlugt) [inline src/workloads/scale/ContentionTTLDeletions.yml:31]: This is the default setting in sys-perf now (on the variants suffixed with 2022-11), so I'm not sure that we need to include it here

File: src/workloads/scale/ContentionTTLDeletions.yml [hunk 3]
```diff
@@ -63,38 +51,183 @@ Actors:
...
+        Collection: *otherColl
+        Operations:
+        - OperationName: insertOne
+          OperationCommand:
+            Document: *LargeDoc
+          OperationOptions:
+            WriteConcern: *WMajorityJFalse
+
+- Name: InsertTTLColl-Small
+  Type: RunCommand
```
Comment (brad-devlugt) [inline src/workloads/scale/ContentionTTLDeletions.yml:194]: What are your thoughts on using the CrudActor for these two as well to be consistent with the earlier actors?

File: src/workloads/scale/ContentionTTLDeletions.yml (hunk 1 above)
Comment (brad-devlugt) [reply src/workloads/scale/ContentionTTLDeletions.yml:37]: Sounds good!

File: src/workloads/scale/ContentionTTLDeletions.yml (hunk 2 above)
Comment (louiswilliams) [reply src/workloads/scale/ContentionTTLDeletions.yml]: I don't think we need to test j:false

**Review by brad-devlugt — DISMISSED** [review]
LGTM with a few patch builds to see how noisy the new test is

**Comment by haleyConnelly** [comment]
I set up 4 patches on 3 node replica sets (both feature flags and non)

- [patch 0](https://spruce.mongodb.com/version/63bc49fe61837d672d78ef7e/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)
- [patch 1](https://spruce.mongodb.com/version/63bc4ace57e85a0faa9de2ab/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)
- [patch 2](https://spruce.mongodb.com/version/63bc4b2e3e8e863ed9aa85c8/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)
- [patch 3](https://spruce.mongodb.com/version/63bc4b7f3e8e863ed9aa86cd/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)

**Comment by haleyConnelly** [comment]
After running 4 runs of the performance workload, we noticed quite a bit of noise.  

Additionally, the workload is no longer consistently contentious with > 0 write out tickets at any given time. 

Originally, this workload was run as a single node replica set to bypass flow control when performance testing did not use w:majority write concern as default. In those runs, it was also consistently contentious.

We're going to run another round of performance tests, where the workload is only run on single node replica sets. Hopefully, this will reduce the noise and maintain the system in a constant state of overload.

**Comment by haleyConnelly** [comment]
https://github.com/mongodb/genny/pull/805

**Comment by haleyConnelly** [comment]
I suspect if we separate the admin commands and index builds of 'other coll' in phase 1, we can a less noisy phase 1

**Review by louiswilliams — APPROVED** [review]

**Review by haleyConnelly — COMMENTED** [review]
The final version of the test will generate noise in set up that we do not care about. 
<google-sheets-html-origin>

test | measurement | Value 0 | Value 1 | Value 2 | Value 3 | Noise Comp
-- | -- | -- | -- | -- | -- | --
UpdateOtherColl.updateOne | OperationThroughput | 1079.043287 | 1079.345765 | 1090.260407 | 1029.457636 | 5.63%
UpdateOtherColl.updateOne | Latency50thPercentile | 199022061 | 199905307 | 198950055 | 208958597 | 5.02%
UpdateOtherColl.updateOne | Latency99thPercentile | 859647668.3 | 809544078.6 | 849522344.4 | 959162718.2 | 17.51%
UpdateOtherColl.Crud | OperationThroughput | 1079.043287 | 1079.345765 | 1090.260407 | 1029.457636 | 5.63%
UpdateOtherColl.Crud | Latency50thPercentile | 199057590 | 199944349 | 198989181 | 208997692 | 5.02%
UpdateOtherColl.Crud | Latency99thPercentile | 859704530.3 | 809614843.6 | 849637067.3 | 959199813.1 | 17.50%
RemoveOtherColl.deleteOne | OperationThroughput | 1044.098166 | 1042.275913 | 1072.621629 | 990.3584116 | 7.89%
RemoveOtherColl.deleteOne | Latency50thPercentile | 202976445.5 | 204339726 | 200501059 | 213888319 | 6.57%
RemoveOtherColl.deleteOne | Latency99thPercentile | 905299148.5 | 864242090.6 | 889568742.4 | 996743198.9 | 14.76%
RemoveOtherColl.Crud | OperationThroughput | 1044.098166 | 1042.275913 | 1072.621629 | 990.3584116 | 7.89%
RemoveOtherColl.Crud | Latency50thPercentile | 203007011 | 204370690 | 200535146 | 213916949 | 6.57%
RemoveOtherColl.Crud | Latency99thPercentile | 905364470.6 | 864297733.8 | 889610235.1 | 996877151.5 | 14.77%
InsertTTLColl-Small.insertOne | OperationThroughput | 2195.127648 | 2195.050681 | 2161.112705 | 2090.683311 | 4.80%
InsertTTLColl-Small.insertOne | Latency50thPercentile | 195449414 | 195983703 | 198945075 | 205858126 | 5.27%
InsertTTLColl-Small.insertOne | Latency99thPercentile | 841090118.3 | 798143861.1 | 837487816.3 | 946138796.4 | 17.63%
InsertTTLColl-Small.Crud | OperationThroughput | 2195.127648 | 2195.050681 | 2161.112705 | 2090.683311 | 4.80%
InsertTTLColl-Small.Crud | Latency50thPercentile | 195516675.5 | 196054891 | 199014219 | 205921087 | 5.27%
InsertTTLColl-Small.Crud | Latency99thPercentile | 841158474.7 | 798213062.3 | 837563844.1 | 946206406.1 | 17.63%
InsertTTLColl-Large.insertOne | OperationThroughput | 1712.478815 | 1709.807423 | 1669.219763 | 1678.210173 | 2.55%
InsertTTLColl-Large.insertOne | Latency50thPercentile | 251131684 | 253423677 | 254803595 | 252391451.5 | 1.45%
InsertTTLColl-Large.insertOne | Latency99thPercentile | 961789281.7 | 909710084.4 | 974254100.9 | 1075739147 | 17.15%
InsertTTLColl-Large.Crud | OperationThroughput | 1712.544584 | 1709.807423 | 1669.219763 | 1678.235376 | 2.56%
InsertTTLColl-Large.Crud | Latency50thPercentile | 253015620 | 255251778 | 256802684 | 254303842.5 | 1.49%
InsertTTLColl-Large.Crud | Latency99thPercentile | 965347293.2 | 915153094 | 977094693.5 | 1079725921 | 16.94%
InsertOtherCollSmall.insertOne | OperationThroughput | 533.1689347 | 534.0612148 | 529.6778227 | 511.7045642 | 4.21%
InsertOtherCollSmall.insertOne | Latency50thPercentile | 201149044.5 | 201723424 | 203367678 | 210918287.5 | 4.82%
InsertOtherCollSmall.insertOne | Latency99thPercentile | 855495855.1 | 808358784.3 | 856360225.5 | 963352883.1 | 18.11%
InsertOtherCollSmall.Crud | OperationThroughput | 533.1689347 | 534.0612148 | 529.6866534 | 511.7045642 | 4.21%
InsertOtherCollSmall.Crud | Latency50thPercentile | 201217068.5 | 201790498 | 203440595 | 210984526.5 | 4.82%
InsertOtherCollSmall.Crud | Latency99thPercentile | 855564134.3 | 808426521.6 | 856438395 | 963446015.9 | 18.11%
InsertOtherCollLarge.insertOne | OperationThroughput | 423.3883273 | 422.538291 | 413.8286395 | 414.0644025 | 2.29%
InsertOtherCollLarge.insertOne | Latency50thPercentile | 252985394 | 255486041 | 256997789 | 255334566 | 1.57%
InsertOtherCollLarge.insertOne | Latency99thPercentile | 970937165.8 | 924451353.3 | 985127611.3 | 1085656562 | 16.48%
InsertOtherCollLarge.Crud | OperationThroughput | 423.3897395 | 422.538291 | 413.8293295 | 414.0685463 | 2.29%
InsertOtherCollLarge.Crud | Latency50thPercentile | 254743505 | 257262201 | 258921818 | 257198703 | 1.62%
InsertOtherCollLarge.Crud | Latency99thPercentile | 975315409.9 | 929926089.6 | 989133524.3 | 1088990051 | 16.19%
CreateTTLCollection.TotalBulkInsert | OperationThroughput | 1 | 1 | 1 | 1 | 0.00%
CreateTTLCollection.TotalBulkInsert | Latency50thPercentile | 107585698 | 134685916 | 118011324 | 98747226 | 31.86%
CreateTTLCollection.TotalBulkInsert | Latency99thPercentile | 107585698 | 134685916 | 118011324 | 98747226 | 31.86%
CreateTTLCollection.IndividualBulkInsert | OperationThroughput | 1 | 1 | 1 | 1 | 0.00%
CreateTTLCollection.IndividualBulkInsert | Latency50thPercentile | 107227705 | 134351427 | 116667580 | 98127910 | 32.36%
CreateTTLCollection.IndividualBulkInsert | Latency99thPercentile | 107227705 | 134351427 | 116667580 | 98127910 | 32.36%
CreateTTLCollection.IndexBuild | OperationThroughput | 1 | 1 | 1 | 1 | 0.00%
CreateTTLCollection.IndexBuild | Latency50thPercentile | 134823387 | 135910742 | 131450006 | 59719515 | 57.23%
CreateTTLCollection.IndexBuild | Latency99thPercentile | 134823387 | 135910742 | 131450006 | 59719515 | 57.23%
CreateIndexesOtherColl.DatabaseOperation.1.1 | OperationThroughput | 1 | 1 | 1 | 1 | 0.00%
CreateIndexesOtherColl.DatabaseOperation.1.1 | Latency50thPercentile | 107271766 | 134599420 | 116727034 | 126538358 | 22.47%
CreateIndexesOtherColl.DatabaseOperation.1.1 | Latency99thPercentile | 107271766 | 134599420 | 116727034 | 126538358 | 22.47%
ConfigureParameters.DatabaseOperation.2.2 | OperationThroughput | 2 | 2000 | 2000 | 2 | 199.60%
ConfigureParameters.DatabaseOperation.2.2 | Latency50thPercentile | 36493517 | 42781695.5 | 41604801.5 | 37451789.5 | 15.91%
ConfigureParameters.DatabaseOperation.2.2 | Latency99thPercentile | 72776977 | 85369617 | 83001975 | 74709341 | 15.97%




**We can expect large variations from the following**
ConfigureParameters -- Sets ttlSleepSecs and number of write out transactions. 
CreateIndexesOtherColl -- Creates the 'otherColl' and its indexes. 
CreateTTLCollection -- Creates the TTL Collection, its index, and inserts a single document.

**Any 99th Latency Percentile Metric will likely have a stable region accounting for at least ~15% noise**
50th Latency Percentiles and Operation Throughput measurements should fall within a <10% noise stable region

**Review by brad-devlugt — APPROVED** [review]

**Comment by haleyConnelly** [comment]
evergreen merge

**Pull request merged**